yt/frontends/nc4_cm1/data_structures.py

```diff
@@ -0,0 +1,233 @@
+        self.time_unit = self.quan(1.0, "s")
+        self.velocity_unit = self.quan(1.0, "m/s")
+        self.time_unit = self.quan(1.0, "s")
+
+    def _parse_parameter_file(self):
+        # This needs to set up the following items.  Note that these are all
+        # assumed to be in code units; domain_left_edge and domain_right_edge
+        # will be converted to YTArray automatically at a later time.
+        # This includes the cosmological parameters.
+        #
+        #   self.unique_identifier      <= unique identifier for the dataset
+        #                                  being read (e.g., UUID or ST_CTIME)
+        self.unique_identifier = int(os.stat(self.parameter_filename)[stat.ST_CTIME])
+        self.parameters = {}  # code-specific items
+        with self._handle.open_ds() as ds:
+            dims = [ds.dimensions[i].size for i in ["xh", "yh", "zh"]]
+            coords = {i: ds.variables[i][:] for i in ["xh", "yh", "zh"]}
+
+            # TO DO: eneralize this to be coordiante variable name agnostic in order to
+            # make useful for WRF or climate data. For now, we're hard coding for CM1
+            # specifically and have named the classes appropriately
+            xh, yh, zh = [ds.variables[i][:] for i in ["xh", "yh", "zh"]]
+            self.domain_left_edge = np.array(
+                [xh.min(), yh.min(), zh.min()], dtype="float64"
+            )
+            self.domain_right_edge = np.array(
+                [xh.max(), yh.max(), zh.max()], dtype="float64"
+            )
+
+            # loop over the variable names in the netCDF file, record only those on the
+            # "zh","yh","xh" grid.
+            varnames = []
+            for key in ds.variables.keys():
+                vardims = ds.variables[key].dimensions
+                if all(x in vardims for x in ["time", "zh", "yh", "xh"]) is True:
+                    varnames.append(key)
+            self.parameters["variable_names"] = varnames
+            self.parameters["lofs_version"] = ds.cm1_lofs_version
+            self.parameters["is_uniform"] = ds.uniform_mesh
+            self.current_time = ds.variables["time"][:][0]
+
+            # record the dimension metadata
+            dim_info = OrderedDict()
+            for dim, meta in ds.dimensions.items():
+                dim_info[dim] = meta.size
+            self.parameters["dimensions"] = dim_info
+
+        self.parameters["coords"] = coords
+
+        self.dimensionality = 3
+        self.domain_dimensions = np.array(dims, dtype="int64")
+        self.periodicity = (False, False, False)
+        self.parameters["time"] = self.current_time
+
+        # Set cosmological information to zero for non-cosmological.
+        self.cosmological_simulation = 0.0
+        self.current_redshift = 0.0
+        self.omega_lambda = 0.0
+        self.omega_matter = 0.0
+        self.hubble_constant = 0.0
+
+    @classmethod
+    def _is_valid(self, *args, **kwargs):
```

Comment:
This is a class method
```suggestion
    def _is_valid(cls, *args, **kwargs):
```